src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -192,17 +192,21 @@ final case class EServicesApiServiceImpl(
 
   private def getProducerEServicesIds(producerId: UUID, consumersIds: List[String])(implicit
     contexts: Seq[(String, String)]
-  ): Future[List[UUID]] =
-    Future
-      .traverse(consumersIds)(consumerId =>
-        // TODO This call will be paginated
-        agreementProcessService.getAgreements(
-          producerId = producerId.toString.some,
-          consumerId.some,
-          states = List(AgreementProcess.AgreementState.ACTIVE, AgreementProcess.AgreementState.SUSPENDED)
+  ): Future[List[UUID]] = {
+    for {
+      agreements <- Future
+        .traverse(consumersIds)(consumerId =>
+          agreementProcessService.getAgreements(
+            consumersIds = Seq(UUID.fromString(consumerId)),
+            producersIds = Seq(producerId),
+            states = List(AgreementProcess.AgreementState.ACTIVE, AgreementProcess.AgreementState.SUSPENDED)
+          )
```

Comment:
Here we do not know how many agreements we will receive, it is required to paginate the results and keep asking until we receive them all.

You can take a look [here](https://github.com/pagopa/interop-be-attribute-registry-management/blob/1.0.x/src/main/scala/it/pagopa/interop/attributeregistrymanagement/api/impl/AttributeApiServiceImpl.scala#L145) for an idea on how to implement it

Reply:
Meglio parlarne direttamente